### PR TITLE
Fixed a bug in Eigenvalue executioner when free_power_iterations=0 

### LINKED ISSUE #17026: Fixed a bug in eigen solver when free_power_iterations = 0
## Bug Description

When `free_power_iterations = 0`, PETSc solver options were not setup correctly.

```
Framework Information:
MOOSE Version:           git commit abf6700985 on 2021-02-16
LibMesh Version:         c70ab40900490f44391f1b891c8c3731c45f0cf1
PETSc Version:           3.14.2
SLEPc Version:           3.14.0
Current Time:            Tue Feb 16 17:25:44 2021
Executable Timestamp:    Tue Feb 16 17:23:27 2021

Parallelism:
  Num Processors:          1
  Num Threads:             1

Mesh: 
  Parallel Type:           replicated
  Mesh Dimension:          2
  Spatial Dimension:       2
  Nodes:                   
    Total:                 81
    Local:                 81
  Elems:                   
    Total:                 64
    Local:                 64
  Num Subdomains:          1
  Num Partitions:          1

Nonlinear System:
  Num DOFs:                81
  Num Local DOFs:          81
  Variables:               "u" 
  Finite Element Types:    "LAGRANGE" 
  Approximation Orders:    "FIRST" 

Execution Information:
  Executioner:             Eigenvalue
  Solver Mode:             Preconditioned JFNK

 Nonlinear Newton iteration starts
[0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[0]PETSC ERROR: No support for this operation for this object type
[0]PETSC ERROR: More than one iteration is not allowed for Newton eigensolver (SNES)
[0]PETSC ERROR: See https://www.mcs.anl.gov/petsc/documentation/faq.html for trouble shooting.
[0]PETSC ERROR: Petsc Release Version 3.14.2, unknown 
[0]PETSC ERROR: ../../../../moose_test-dbg on a arch-moose named Fandes-Mac-Pro.local by kongf Tue Feb 16 17:25:43 2021
[0]PETSC ERROR: Configure options --with-make-np=24 --download-hypre=1 --with-debugging=no --with-shared-libraries=1 --download-fblaslapack=1 --download-metis=1 --download-ptscotch=1 --download-parmetis=1 --download-superlu_dist=1 --download-mumps=1 --download-strumpack=1 --download-scalapack=1 --download-slepc=1 --with-mpi=1 --with-cxx-dialect=C++11 --with-fortran-bindings=0 --with-sowing=0
[0]PETSC ERROR: #1 EPSSetUp_Power() line 109 in /Users/kongf/projects/moose2/petsc/arch-moose/externalpackages/git.slepc/src/eps/impls/power/power.c
[0]PETSC ERROR: #2 EPSSetUp() line 315 in /Users/kongf/projects/moose2/petsc/arch-moose/externalpackages/git.slepc/src/eps/interface/epssetup.c
[0]PETSC ERROR: #3 EPSSolve() line 136 in /Users/kongf/projects/moose2/petsc/arch-moose/externalpackages/git.slepc/src/eps/interface/epssolve.c
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
[unset]: write_line error; fd=-1 buf=:cmd=abort exitcode=1
```

## Steps to Reproduce

`../../../../moose_test-dbg  -i ne_ic.i  Executioner/free_power_iterations=0`

## Impact
Improvement

### PR DESCRIPTION
**Pull request by fdkong**
Closes #17026

### CODE REVIEW

**Comment by fdkong** [comment]
@YaqiWang Could you check if it works in griffin? And any other issues?

**Comment by YaqiWang** [comment]
This works fine for Griffin. But why the old petsc produces different (better) convergence to the correct eigenvalue? I am concerned. I have noticed that the new eigenvalue executioner requires more nonlinear iterations than the old eigenvalue executioner.

**Review by YaqiWang — CHANGES_REQUESTED** [review]
Looks good to me. You said old slepc always does free power iteration internally. How many of that? We do not have a command-line option to control that number?

File: test/tests/problems/eigen_problem/initial_condition/tests [hunk 1]
```diff
@@ -11,6 +11,18 @@
...
+  [./newton_no_free_power]
+    type = 'CSVDiff'
+    input = 'ne_ic.i'
+    csvdiff = 'no_free_power_eigenvalues_0001.csv'
+    # Old slepc will do free power automatically, and we can not control that
+    # But anyway we already require 
+    slepc_version = '>=3.13.0'
+    cli_args = 'Executioner/free_power_iterations=0 Outputs/file_base=no_free_power'
+    issues = '#17026'
+    requirement = "The system shall work correctly when use no free power iteration."
```
Comment (loganharbour) [inline test/tests/problems/eigen_problem/initial_condition/tests:23]:
```suggestion
    requirement = "The system shall support eigenvalue execution without free power iterations."
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

**Review by YaqiWang — DISMISSED** [review]

**Comment by fdkong** [comment]
@loganharbour Ready for review again

**Review by loganharbour — APPROVED** [review]

**Pull request merged**